library/src/main/java/com/squareup/timessquare/MonthView.java

```diff
@@ -48,18 +50,23 @@ public static MonthView create(ViewGroup parent, LayoutInflater inflater,
       view.setDayBackground(dayBackgroundResId);
     }
 
-    final int originalDayOfWeek = today.get(Calendar.DAY_OF_WEEK);
-
     view.isRtl = isRtl(locale);
     view.locale = locale;
     int firstDayOfWeek = today.getFirstDayOfWeek();
     final CalendarRowView headerRow = (CalendarRowView) view.grid.getChildAt(0);
-    for (int offset = 0; offset < 7; offset++) {
-      today.set(Calendar.DAY_OF_WEEK, getDayOfWeek(firstDayOfWeek, offset, view.isRtl));
-      final TextView textView = (TextView) headerRow.getChildAt(offset);
-      textView.setText(weekdayNameFormat.format(today.getTime()));
+
+    if (displayWeekView) {
+      final int originalDayOfWeek = today.get(Calendar.DAY_OF_WEEK);
+      for (int offset = 0; offset < 7; offset++) {
+        today.set(Calendar.DAY_OF_WEEK, getDayOfWeek(firstDayOfWeek, offset, view.isRtl));
+        final TextView textView = (TextView) headerRow.getChildAt(offset);
+        textView.setText(weekdayNameFormat.format(today.getTime()));
+      }
+      today.set(Calendar.DAY_OF_WEEK, originalDayOfWeek);
+    } else {
+      view.weekView.setVisibility(View.GONE);
```

Comment:
don't you need a corresponding ` view.weekView.setVisibility(View.VISIBLE);`?

Reply:
Since the view is always shown (the xml doesn't have the visibility attribute on false) what I did was just avoiding the set text and go directly to hide the  **day_names_header_row** view. I think that scenario is handled by itself when the attribute is true. Let me know if you think that's not enough 👍